frontend/src/components/pages/profile/Profile.tsx

```diff
@@ -1,10 +1,186 @@
+import { ServiceHandlers } from "../../contexts/contexts";
+import { Theme } from "@material-ui/core/styles";
+import TextField from '@material-ui/core/TextField';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexWrap: 'wrap',
+      marginTop: '100px',
+      marginLeft: theme.spacing(4),
+    },
+    '& input:invalid + fieldset': {
+      borderColor: 'red',
+      borderWidth: 2,
+    },
+    textField: {
+      marginLeft: theme.spacing(1),
+      marginRight: theme.spacing(1),
+      width: '25ch',
+    },
+  }),
+);
+
+interface ProfileProps {
+  profileHandlerService: ProfileHandlerService;
+}
+
+interface ProfileState {
+  personProps: PersonProps;
+}
+
+export default function Profile(props: ProfileProps, state: ProfileState) {
+
+  const classes = useStyles();
+
+  /** 
+   * ServiceHandlers is an object containing various TS Handlers and provides
+   * functionality to communicate data from the frontend to the backend
+   */
+  const serviceHandlers = React.useContext(ServiceContext);
+  const profileHandlerService = serviceHandlers.profileHandlerService;
+  
+  const [person, setPerson] = React.useState<PersonProps|undefined>(undefined);
+  const [profileId, setProfileId] = React.useState<string>(undefined);
+  const [submitSuccess, setSubmitSuccess] = React.useState<Boolean|undefined>(undefined);
+  
+  React.useEffect(() => {
+    (async() => {
+      const personPromise = await profileHandlerService.getPerson(USER_ID);
+      setPerson(personPromise);
+    })();
+  }, [profileId]); /** Re-renders Profile only when the userId changes */
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, email: event.target.value});
+  }
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, nickname: event.target.value});
+  }
+
+  const handlePronounsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPerson({...person, pronouns: event.target.value});
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    (async() => {
+      const success = (person.email === "" || person.nickname === "")
+        ? false /** At least one required field is missing */
+        : await profileHandlerService.updatePerson(person);
+      setSubmitSuccess(success);
+    })();
+  };
 
-export const Profile = () => {
   return (
-    <div>
-      <h2>Profile</h2>
-      <p>This is the Profile page.</p>
+    <div className={classes.root}>
+      <form>
+        <div>
+          <TextField
+            fullWidth
+            helperText="to be displayed around CoffeeHouse"
+            id="nickname"
+            InputLabelProps={{
+              shrink: true,
+            }}
+            label="Nickname"
+            margin="normal"
+            onChange={handleNameChange}
+            placeholder="Nickname"
+            style={{ margin: 8 }}
```

Comment:
These are all `margin: 8` so set the style in the `useStyle` part, not the template